### PR TITLE
Order of different types of sorting

### LINKED ISSUE #2617: Order of different types of sorting

### PR DESCRIPTION
**Pull request by LamaEats**
Now sort parameters will be apply in select order and have separate list for selected parameters

## PR includes

- [x] Bug Fix
- [x] Refactor
- [x] Feature

## Related issues

Resolve #2617 

## QA Instructions, Screenshots, Recordings

image

### CODE REVIEW

File: src/components/SortList/SortList.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        [],
+    );
+
+    const itemsToRender: SingleSortItem[] = useMemo(() => {
+        return (Object.entries(sortItems) as Array<[SortableProps, string]>).map(([id, title]) => ({
+            id,
+            title,
+            dir: null,
+        }));
+    }, [sortItems]);
```
Comment (asabotovich) [inline src/components/SortList/SortList.tsx:42]: we can return both values from one `useMemo`

`const {sortItems, itemsToRender} = useMemo(() => ..., []) ;`

File: src/hooks/useUrlFilterParams.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ export interface FilterQueryState {
     participant: string[];
     project: string[];
     query: string;
-    sort: { [K in SortableProps]?: SortDirection };
+    sort: Array<{ key: SortableProps; dir: Exclude<SortDirection, null> }>;
```
Comment (asabotovich) [inline src/hooks/useUrlFilterParams.ts:34]: unnecessary Exclude

File: trpc/queries/goalV2.ts [hunk 3]
```diff
@@ -59,6 +57,10 @@ const mapSortParamsToTableColumns = (sort: QueryWithFilters['sort']): Array<Orde
     });
 };
 
+/**
+ * @deprecated
+ * DONT USE THIS QUERY
+ */
 export const getGoalList = (params: GoalQueryParams) => {
```
Comment (asabotovich) [inline trpc/queries/goalV2.ts:64]: Can we remove it? I can't find usage.

Comment (LamaEats) [reply trpc/queries/goalV2.ts:64]: Maybe not in this pr?)

Its will use in future

**Review by asabotovich — APPROVED** [review]

**Review by tsumo — APPROVED** [review]

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**